### ISSUE TITLE
Clutter causing Unity Game Crashes - Severe Issue

### ISSUE THREAD

**Issue by ValhallaCommunity** [plumga/Clutter#7]
**__Bug Issue Report__**
**Mod Name**: Clutter
**Issue**: Triggering Unity Errors causing game to crash. 
**Severity**: Critical
**Likelihood**: High
**Reproducibility (100%)**: 

**Description**: When players approach builds that include pieces from clutter, their game is crashing. These are build pieces that were on the world prior to the recent Valheim patch. There has been wear and tear on these items.

**Steps to reproduce**
1) Kept Clutter mod in our world. Crashes happen at locations where clutter build pieces / objects are included. This is happening for every player that comes into contact with the an objects wear n tear condition.
2) Once I tested removing Clutter mod and re-adding about 5 times, the unity error crash happens when Clutter is included on server and you are near a specific item.

Note: I tested all mods and can only reproduce this unity game crash error with clutter since the Valheim game patch update to 0.214.2.

**Error Log #1 (Unity Crash Log):**
========== OUTPUTTING STACK TRACE ==================

```
0x00007FFC3D4E9226 (UnityPlayer) UnityMain
0x00007FFC3D4EA40B (UnityPlayer) UnityMain
0x00007FFC3D8B45AA (UnityPlayer) UnityMain
0x000001765A9790D9 (Mono JIT Code) (wrapper managed-to-native) UnityEngine.GameObject:SetActive (UnityEngine.GameObject,bool)
0x0000017B0326AE3B (Mono JIT Code) WearNTear:SetHealthVisual (single,bool)
0x0000017B0326AC2B (Mono JIT Code) WearNTear:UpdateVisual (bool)
0x0000017B0328965B (Mono JIT Code) WearNTear:UpdateWear ()
0x0000017B02E38BEB (Mono JIT Code) WearNTearUpdater/<UpdateWear>d__1:MoveNext ()
0x000001763795860C (Mono JIT Code) UnityEngine.SetupCoroutine:InvokeMoveNext (System.Collections.IEnumerator,intptr)
0x000001763246A417 (Mono JIT Code) (wrapper runtime-invoke) <Module>:runtime_invoke_void_object_intptr (object,intptr,intptr,intptr)
0x00007FFC61E4F1E0 (mono-2.0-bdwgc) mono_get_runtime_build_info
0x00007FFC61DD2AC2 (mono-2.0-bdwgc) mono_perfcounters_init
0x00007FFC61DDBB1F (mono-2.0-bdwgc) mono_runtime_invoke
0x00007FFC3D83CFAD (UnityPlayer) UnityMain
0x00007FFC3D83A0A2 (UnityPlayer) UnityMain
0x00007FFC3D81D555 (UnityPlayer) UnityMain
0x00007FFC3D595E24 (UnityPlayer) UnityMain
0x00007FFC3D6C80CA (UnityPlayer) UnityMain
0x00007FFC3D6C8170 (UnityPlayer) UnityMain
0x00007FFC3D6CC408 (UnityPlayer) UnityMain
  ERROR: SymGetSymFromAddr64, GetLastError: 'Attempt to access invalid address.' (Address: 00007FFC3D48C6BA)
0x00007FFC3D48C6BA (UnityPlayer) (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 'Attempt to access invalid address.' (Address: 00007FFC3D48AB8B)
0x00007FFC3D48AB8B (UnityPlayer) (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 'Attempt to access invalid address.' (Address: 00007FFC3D48FD02)
0x00007FFC3D48FD02 (UnityPlayer) (function-name not available)
0x00007FFC3D490C8B (UnityPlayer) UnityMain
  ERROR: SymGetSymFromAddr64, GetLastError: 'Attempt to access invalid address.' (Address: 00007FF70A5011F2)
0x00007FF70A5011F2 (valheim) (function-name not available)
0x00007FFCA62F7614 (KERNEL32) BaseThreadInitThunk
0x00007FFCA7F826A1 (ntdll) RtlUserThreadStart
```

========== END OF STACKTRACE ===========

**Error Log #2**
One time, I was able to get this error from a client log prior to a crash. But client and server logs has not logged the clutter crash

```
[Error  : Unity Log] NullReferenceException
Stack trace:
WearNTear.SetHealthVisual (System.Single health, System.Boolean triggerEffects) (at <afd4e3f8ab8b460f8ff25d476fbd573e>:0)
WearNTear.UpdateVisual (System.Boolean triggerEffects) (at <afd4e3f8ab8b460f8ff25d476fbd573e>:0)
WearNTear.UpdateWear () (at <afd4e3f8ab8b460f8ff25d476fbd573e>:0)
WearNTearUpdater+<UpdateWear>d__1.MoveNext () (at <afd4e3f8ab8b460f8ff25d476fbd573e>:0)
UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <c9fb31c2eee14ce4aeb36e4c61e154d8>:0)
```

Any assistance would be greatly appreciated!!

**Comment by plumga** [comment]
working on the fix, thanks for the report! In the meantime, use this to fix the crashing issue as a temporary workaround:
https://valheim.thunderstore.io/package/Azumatt/Clutter_Sign_Component_Fix/

**Comment by plumga** [comment]
fixed in V0.1.4

**Comment by ValhallaCommunity** [comment]
Confirming fix for notes! Thank you Plum! I'll retest with AzuSigns in the near future! Really appreciate the help and have a good week at the big job!!
